oml/datasets/audios.py

```diff
@@ -0,0 +1,353 @@
+        input_frame_offset_key: str = INPUT_FRAME_OFFSET_KEY,
+        index_key: str = INDEX_KEY,
+        labels_key: str = LABELS_KEY,
+    ):
+        assert all(x in df.columns for x in (LABELS_COLUMN, PATHS_COLUMN))
+        self.labels_key = labels_key
+        self.df = df
+
+        extra_data = extra_data or dict()
+
+        if CATEGORIES_COLUMN in df.columns:
+            extra_data[CATEGORIES_COLUMN] = df[CATEGORIES_COLUMN].copy()
+
+        if SEQUENCE_COLUMN in df.columns:
+            extra_data[SEQUENCE_COLUMN] = df[SEQUENCE_COLUMN].copy()
+
+        super().__init__(
+            paths=df[PATHS_COLUMN],
+            dataset_root=dataset_root,
+            extra_data=extra_data,
+            sr=sr,
+            num_seconds=num_seconds,
+            num_channels=num_channels,
+            input_tensors_key=input_tensors_key,
+            input_frame_offset_key=input_frame_offset_key,
+            index_key=index_key,
+        )
+
+    def __getitem__(self, item: int) -> Dict[str, Any]:
+        data = super().__getitem__(item)
+        data[self.labels_key] = self.df.iloc[item][LABELS_COLUMN]
+        return data
+
+    def get_labels(self) -> NDArray[np.int64]:
+        return np.array(self.df[LABELS_COLUMN])
+
+    def get_label2category(self) -> Optional[Dict[int, Union[str, int]]]:
+        if CATEGORIES_COLUMN in self.df.columns:
+            label2category = dict(zip(self.df[LABELS_COLUMN], self.df[CATEGORIES_COLUMN]))
+        else:
+            label2category = None
+
+        return label2category
+
+
+class AudioQueryGalleryDataset(AudioBaseDataset, IQueryGalleryDataset):
+    """
+    The **non-annotated** dataset of audios having `query`/`gallery` split.
+    To perform `1 vs rest` validation, where a query is evaluated versus the whole validation dataset
+    (except for this exact query), you should mark the item as ``is_query == True`` and ``is_gallery == True``.
+
+    """
+
+    def __init__(
+        self,
+        df: pd.DataFrame,
+        dataset_root: Optional[Union[str, Path]] = None,
+        extra_data: Optional[Dict[str, Any]] = None,
+        sr: int = DEFAULT_SAMPLE_RATE,
+        num_seconds: float = DEFAULT_DURATION,
+        num_channels: int = DEFAULT_NUM_CHANNELS,
+        input_tensors_key: str = INPUT_TENSORS_KEY,
+        input_frame_offset_key: str = INPUT_FRAME_OFFSET_KEY,
+        index_key: str = INDEX_KEY,
+    ):
+        assert all(x in df.columns for x in (IS_QUERY_COLUMN, IS_GALLERY_COLUMN, PATHS_COLUMN))
+        self.df = df
+
+        super().__init__(
+            paths=df[PATHS_COLUMN],
+            dataset_root=dataset_root,
+            extra_data=extra_data,
+            sr=sr,
+            num_seconds=num_seconds,
+            num_channels=num_channels,
+            input_tensors_key=input_tensors_key,
+            input_frame_offset_key=input_frame_offset_key,
+            index_key=index_key,
+        )
+
+    def get_query_ids(self) -> LongTensor:
```

Comment:
please, don't copy my mistake :) 

take a look at the [fix](https://github.com/OML-Team/open-metric-learning/commit/0e3776f8507bb3ed95bcdbd7ebfdb3515e724361)